relay-server/src/extractors/start_time.rs

```diff
@@ -19,6 +19,26 @@ impl StartTime {
     pub fn into_inner(self) -> Instant {
         self.0
     }
+
+    /// Returns the [`StartTime`] corresponding to provided timestamp.
+    pub fn from_timestmap(timestamp: u64) -> Self {
+        let ts = Duration::from_millis(timestamp);
+
+        let elapsed = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap_or_default()
+            .checked_sub(ts)
+            .unwrap_or_default();
+
+        let start_time = Instant::now()
+            // Subtract the elapsed time from the current `Instant` to get the timestamp
+            // of the start time.
+            .checked_sub(elapsed)
+            // If we fail to get the `Instant` from the timestamp, we fallback to `now()`.
+            .unwrap_or_else(Instant::now);
+
+        Self(start_time)
+    }
```

Comment:
Instead of this function, can we use `UnixTimestamp::to_instant`?

https://github.com/getsentry/relay/blob/e5d7d5b64d9a46897729b2bac500713310f19bc0/relay-common/src/time.rs#L155

Reply:
This will introduce another layer of indirection:
* this wrapper assumes the internal timestamp in seconds, which will have to be fixed
* this also will require more additions to the `impl` of `UnixTimestamp` than this PR already makes

I can work on followup to improve `UnixTimestamp` to handle better precision and then use this wrapper inside of this function (or instead).  What do you think?

Reply:
Ah right, `UnixTimestamp` counts seconds. Never mind.